packages/vaadin-virtual-list/src/vaadin-virtual-list.d.ts

```diff
@@ -59,6 +59,16 @@ declare class VirtualListElement extends ElementMixin(ThemableMixin(HTMLElement)
    * Scroll to a specific index in the virtual list.
    */
   scrollToIndex(index: number): void;
+
+  /**
+   * Gets the index of the first visible item in the viewport.
+   */
+  firstVisibleIndex: number;
```

Comment:
Let's indicate this is a read-only property:

```suggestion
  readonly firstVisibleIndex: number;
```